### PR TITLE
Add compile time guard for fullscreen video dismissal gestures

### PR DESCRIPTION
**Pull request by grorg**
#### f0703be2f94ac0255a2231e2eed54bc59b575c52
<pre>
Add compile time guard for fullscreen video dismissal gestures
<a href="https://bugs.webkit.org/show_bug.cgi?id=242222">https://bugs.webkit.org/show_bug.cgi?id=242222</a>

Reviewed by Antoine Quint and Devin Rousso.

Add ENABLE(FULLSCREEN_DISMISSAL_GESTURES), which is enabled
on PLATFORM(IOS_FAMILY).

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen]):

Canonical link: <a href="https://commits.webkit.org/252074@main">https://commits.webkit.org/252074@main</a>
</pre>

### CODE REVIEW

**Review by graouts — APPROVED** [review]

File: Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm [hunk 1]
```diff
@@ -550,6 +550,7 @@ - (void)enterFullScreen
     _fullscreenViewController.get().view.frame = _rootViewController.get().view.bounds;
     [self _updateLocationInfo];
 
+#if HAVE(FULLSCREEN_DISMISSAL_GESTURES)
     _startDismissGestureRecognizer = adoptNS([[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(_startToDismissFullscreenChanged:)]);
```
Comment (dcrousso) [inline Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:560]: I don't think guarding this is enough.  There's lots of uses of these ivars in other methods (not to mention the ivars and methods themselves) that we probably want to guard too.

Also, what about `_interactiveDismissTransitionCoordinator`?

**Review by dcrousso — CHANGES_REQUESTED** [review]

File: Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm (hunk 1 above)
Comment (grorg) [reply Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:560]: There were a couple of other places. We didn't really need to guard the ivars or the gesture recognizer calls, but it is cleaner.

File: Source/WTF/wtf/PlatformHave.h [hunk 2]
```diff
@@ -1285,6 +1285,10 @@
 #define HAVE_MEDIA_VOLUME_PER_ELEMENT 1
 #endif
 
+#if PLATFORM(IOS_FAMILY)
+#define HAVE_FULLSCREEN_DISMISSAL_GESTURES 1
```
Comment (dcrousso) [inline Source/WTF/wtf/PlatformHave.h]: I think this needs to be moved to `PlatformEnableCocoa` with a `#if !defined(ENABLE_FULLSCREEN_DISMISSAL_GESTURES)`

File: Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm [hunk 3]
```diff
@@ -550,6 +552,7 @@ - (void)enterFullScreen
     _fullscreenViewController.get().view.frame = _rootViewController.get().view.bounds;
     [self _updateLocationInfo];
 
+#if ENABLE(FULLSCREEN_DISMISSAL_GESTURES)
     _startDismissGestureRecognizer = adoptNS([[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(_startToDismissFullscreenChanged:)]);
```
Comment (dcrousso) [inline Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:560]: we probably could/should also guard `_startToDismissFullscreenChanged:` too while we're doing the rest :)

**Review by dcrousso — APPROVED** [review]

**Pull request merged**